### PR TITLE
Enable runtime closure verification

### LINKED ISSUE #31805: Fix up NETCoreApp closure validation when assembling runtime packages
In a recent official build binlog, it looks like `VerifyClosure` isn't running, and its dependency `GetClosureFiles` doesn't show up at all. This is an important check to ensure shared framework integrity.

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Fixes: https://github.com/dotnet/runtime/issues/31805

Overriding Arcade target to allow closure verification without other validations and to allow us to incorporate a fix for a bug in VerifyClosure. As discovered during fix verification, the task would fail if a destination folder (for DGML file) does not exist - issue in Arcade: https://github.com/dotnet/arcade/issues/6090

### CODE REVIEW

File: src/installer/pkg/projects/Directory.Build.targets
Comment (ericstj) [inline src/installer/pkg/projects/Directory.Build.targets:149]: If this is just temporary rather than forking the whole thing, just have a single target:
```xml
<Target Name="EnsureDepenedencyGraphDir" BeforeTargets="VerifyClosure">
    <MakeDir Directories="$(PackageReportDir)$(Id)" />
</Target>
```

File: src/installer/pkg/projects/netcoreapp/pkg/legacy/Microsoft.NETCore.App.pkgproj
Comment (ericstj) [inline src/installer/pkg/projects/netcoreapp/pkg/legacy/Microsoft.NETCore.App.pkgproj:19]: I don't think we should ignore this one.  We weren't before.
https://github.com/dotnet/corefx/blob/d6302a72f8eafa326d7572a02acf8f3021ebb9e8/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj#L18-L49

We only ignore it when examining closure of standalone-packages.  When examining the closure of netcoreapp we want it to have zero cycles.

Comment (ericstj) [reply src/installer/pkg/projects/netcoreapp/pkg/legacy/Microsoft.NETCore.App.pkgproj:19]: You may encounter a failure, and that should be fixed by adding a netcoreappcurrent configuration for the lib that causes the cycle.

Comment (jkoritzinsky) [reply src/installer/pkg/projects/netcoreapp/pkg/legacy/Microsoft.NETCore.App.pkgproj:19]: I agree with Eric. Ignoring `netstandard` would cause us to miss all of the failures we've been seeing recently use to regressions from having this verification disabled.

File: src/installer/pkg/projects/Directory.Build.targets
Comment (NikolaMilosavljevic) [reply src/installer/pkg/projects/Directory.Build.targets:149]: I actually need to override the target to update it's condition as well, to run if new property is set - `ShouldVerifyClosure`. Currently `VerifyClosure` only runs if `SkipValidatePackage` is false: https://github.com/dotnet/arcade/blob/e11b8486ad155fab13e788a268755c0ca60d3609/src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.packaging.targets#L170

File: src/installer/pkg/projects/netcoreapp/pkg/legacy/Microsoft.NETCore.App.pkgproj
Comment (NikolaMilosavljevic) [reply src/installer/pkg/projects/netcoreapp/pkg/legacy/Microsoft.NETCore.App.pkgproj:19]: Thanks - I'll remove it - I think I've seen issues with netstandard cycle in my Linux build - will double-check.

Comment (NikolaMilosavljevic) [reply src/installer/pkg/projects/netcoreapp/pkg/legacy/Microsoft.NETCore.App.pkgproj]: This is fixed with latest commit.

File: src/installer/pkg/projects/Directory.Build.targets
Comment (ericstj) [reply src/installer/pkg/projects/Directory.Build.targets:149]: Is there a reason we can’t get the arcade update instead?  Would it take too long?  Could you manually update just the package with this fix?

Comment (NikolaMilosavljevic) [reply src/installer/pkg/projects/Directory.Build.targets:149]: I've heard that there is a freeze on whole Arcade update, but I'll ask if a single package would be allowed.

Comment (NikolaMilosavljevic) [reply src/installer/pkg/projects/Directory.Build.targets:149]: Heard back - we can get update in RC2. I will update `VerifyClosure` target in `Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk` in Arcade, and remove the target override from my PR. But, I will pick your suggestion for a new target to create `Reports` directory, which works around an issue in `VerifyClosure` task: https://github.com/dotnet/arcade/issues/6090. The task is in a different Arcade package and they want to minimize the number of changes now.

Comment (NikolaMilosavljevic) [reply src/installer/pkg/projects/Directory.Build.targets:149]: @ericstj I will also move your suggested new target into Arcade next to VerifyClosure target. We can, then, remove this target when VerifyClosure task is fixed - with a single PR in Arcade.

Comment (NikolaMilosavljevic) [reply src/installer/pkg/projects/Directory.Build.targets:149]: Will keep `EnsureDepenedencyGraphDir` in runtime repo - `PackageReportDir` is runtime-specific property, it doesn't make sense for this to be in Arcade.

Comment (NikolaMilosavljevic) [reply src/installer/pkg/projects/Directory.Build.targets]: Arcade fix is in: https://github.com/dotnet/arcade/pull/6101

This (runtime) fix can be merged and closure validation would start working once we pick up new Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk package.

@ericstj @jkoritzinsky this is safe to merge once you approve the changes.

**Comment by ericstj** [comment]
If I understand correctly the current state of this PR is not running the target and you need to include the new package in order to fix the condition.  Is that correct?

**Comment by NikolaMilosavljevic** [comment]
> If I understand correctly the current state of this PR is not running the target and you need to include the new package in order to fix the condition. Is that correct?

Yes, that's correct - this PR needs to be completed by 9/4, but Arcade change will come a bit later.

**Comment by NikolaMilosavljevic** [comment]
> If I understand correctly the current state of this PR is not running the target and you need to include the new package in order to fix the condition. Is that correct?

@ericstj I don't mind copying `VerifyClosure` target from Arcade and removing it later. We already need to remove the other target once `VerifyClosure` task issue is fixed.

If we do this, we ensure that target runs and we're not blocked by package update.

**Comment by ericstj** [comment]
I see, one thing I've done in the past is manually update a single package once signed build is available (which is typically an hour or two after check-in).   Is that not possible?

**Comment by NikolaMilosavljevic** [comment]
> I see, one thing I've done in the past is manually update a single package once signed build is available (which is typically an hour or two after check-in). Is that not possible?

Yeah - I've added that change now.

**Review by ericstj — APPROVED** [review]
This looks good.  Did you confirm the target is running?

**Comment by NikolaMilosavljevic** [comment]
> This looks good. Did you confirm the target is running?

Yes - confirmed in my local build, but will check PR validation build as well, once it completes.

**Comment by NikolaMilosavljevic** [comment]
> This looks good. Did you confirm the target is running?

Yes, the target is running. However, I might need to update this PR. Double-checking right now, but I might need to pick a new version of another package, due to some refactoring done about a month ago, by @mmitche : https://github.com/dotnet/arcade/commit/48646648a006e11c5e0e3696abc7d61f44fc0587#diff-af7c096fa15e403c04f3343d8a820d57

**Comment by NikolaMilosavljevic** [comment]
> > This looks good. Did you confirm the target is running?
> 
> Yes, the target is running. However, I might need to update this PR. Double-checking right now, but I might need to pick a new version of another package, due to some refactoring done about a month ago, by @mmitche : [dotnet/arcade@4864664#diff-af7c096fa15e403c04f3343d8a820d57](https://github.com/dotnet/arcade/commit/48646648a006e11c5e0e3696abc7d61f44fc0587#diff-af7c096fa15e403c04f3343d8a820d57)

Confirmed that the target is running in validation builds.

Also confirmed that dependent intra-Arcade package is properly restored during build.

**Pull request merged**
